temba/reports/views.py

```diff
@@ -0,0 +1,263 @@
+from django.http import HttpResponseRedirect, JsonResponse
+from rest_framework import status
+from rest_framework.views import APIView
+from rest_framework.response import Response as APIResponse
+from smartmin.views import SmartCRUDL, SmartCreateView, SmartTemplateView
+from temba.orgs.views import OrgPermsMixin
+from .models import Report, DataCollectionProcess
+from .tasks import manually_collect_flow_results_data
+from ..contacts.models import ContactGroup
+from ..flows.models import Flow, FlowRunCount
+
+
+class ReportCRUDL(SmartCRUDL):
+    actions = ("create", "delete", "analytics", "charts_data", "update_charts_data")
+    model = Report
+
+    class Create(OrgPermsMixin, SmartCreateView):
+        success_message = ""
+
+        def get(self, request, *args, **kwargs):
+            return HttpResponseRedirect(reverse("reports.report_read"))
+
+        def post(self, request, *args, **kwargs):
+            json_string = request.body
+            user = request.user
+            org = user.get_org()
+
+            try:
+                json_dict = json.loads(json_string)
+            except Exception as e:
+                return JsonResponse(dict(status="error", description="Error parsing JSON: %s" % str(e)), status=400)
+
+            try:
+                report = Report.create_report(org, user, json_dict)
+            except Exception as e:  # pragma: needs cover
+                traceback.print_exc(e)
+                return JsonResponse(dict(status="error", description="Error creating report: %s" % str(e)), status=400)
+
+            return JsonResponse(dict(status="success", description="Report Created", report=report.as_json()))
+
+    class Delete(OrgPermsMixin, SmartTemplateView, APIView):
+        permission = "reports.report_delete"
+
+        def delete(self, request, *args, **kwargs):
+            org = request.user.get_org()
+            report_id = request.data.get("report_id")
+            if all((org, report_id)):
+                report = Report.objects.filter(org=org, id=report_id).first()
+                if report:
+                    report.delete()
+            return APIResponse({}, status=status.HTTP_204_NO_CONTENT)
+
+    class Analytics(OrgPermsMixin, SmartTemplateView):
+        title = "Analytics"
+        permission = "reports.report_read"
+
+        def get_context_data(self, **kwargs):
+            org = self.request.user.get_org()
+            dev_mode = getattr(settings, "EDITOR_DEV_MODE", False)
+            prefix = "http://localhost:3000" if dev_mode else settings.STATIC_URL
+
+            # get our list of assets to include
+            scripts = []
+            styles = []
+
+            if dev_mode:  # pragma: no cover
+                response = requests.get("http://localhost:3000/asset-manifest.json")
+                data = response.json()
+            else:
+                with open("node_modules/@greatnonprofits-nfp/temba-analytics/build/asset-manifest.json") as json_file:
```

Comment:
Add [@greatnonprofits-nfp/temba-analytics](https://www.npmjs.com/package/@greatnonprofits-nfp/temba-analytics) to package.json file

Reply:
Have added a dependency to package.json